proposals/unsigned/unsigned.txt

```diff
@@ -0,0 +1,228 @@
+To: J3                                                     J3/24-XXX
+From:
+Subject: Adding an UNSIGNED type to Fortran
+Date: 2024-October-25
+
+References: 24-116, 24-102, 07-007
+            WG5 N2230 DIN Suggestions for F202Y.pdf
+            WG5 N2142 Fortran 2020 Feature Survey Results 201710.pdf
+
+# 1. Introduction
+
+We propose adding a small set of features for an unsigned data to
+Fortran 202y. Unsigned integers are a basic data type used in many
+programming languages, such as C. They are useful for a range of
+applications, including, but not limited to
+
+- interfacing to C
+- interfacing to the operating system
+- random number generators
+- image processing
+- signal processing
+- hashing
+- cryptography (including multi-precision arithmetic)
+- data compression
+- binary file I/O
+
+Unsigned integers were the fourth most requested item to add to Fortran
+202x in 2017. It is the sixth item on the DIN national body list for
+inclusion in Fortran 202y.
+
+The use cases can be roughly divided into three classes:
+
+- representing unsigned integer
+- bit operations
+- modular arithmetic (2^n for a datatype with n bits)
+
+The two fundamental designs are:
+
+- adding a dedicated type for each use case with the appropriate
+  behavior of aritmetic operators and intrinsic functions on overflow;
+  the different types can possibly just be different kinds for
+  `unsigned(kind=...)`:
+  * `unsigned`: arithmetic operation overflow do not wraparound, or are
+    possibly not even defined
+  * `bits`: bit operations are defined, arithmetic operations do not
+    wraparound or are not defined
+  * `modular`: arithmetic operations wraparound using modular 2^n
+    arithmetic
```

Comment:
There was also discussing of clamped or saturating semantics.  Which would be a bad default, but if you're going to list all of the choices, you missed that one.

Reply:
Good point. Btw, I added the appropriate intrinsics in 0e6544de4a90a5d5874445073341ec4a1f9d95d4.